### PR TITLE
Upgrade SDK dependency to 2.12.0-0

### PR DESCRIPTION
**Pull request by lrhn**
Also depend on package:convert 3.0.0-nullsafety.

### CODE REVIEW

**Comment by mit-mit** [comment]
What would it take to drop the dependency override on `convert`?

**Comment by mit-mit** [comment]
We should update the version (currently `version: 2.2.0-nullsafety-dev`) to use our recommended format, which I believe would be `version: 3.0.0-nullsafety.0`?

**Comment by lrhn** [comment]
I updated the version number to 3.0.0-nullsafety.0.
Now we need dep-overrides for many more packages (all of those directly or transitively depending on crypto <3.0.0)

**Review by jakemac53 — APPROVED** [review]
@natebosch  was assigned this package in the sheet so I would wait for his lgtm that this is ready to publish just to double check, otherwise we should leave the version at `-nullsafety-dev` (or maybe `-nullsafety.0-dev`?

**Comment by lrhn** [comment]
Definitely not ready to publish, so @natebosch, feel free to start from scratch or take over this attempt.

**Comment by jakemac53** [comment]
This could still land if we change the version to `2.2.0-nullsafety.0-dev` in that case, bumping the constraint LGTM

File: pubspec.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 name: crypto
-version: 2.2.0-nullsafety-dev
+version: 2.0.0-nullsafety.0
```
Comment (natebosch) [inline pubspec.yaml:2]:
```suggestion
version: 2.2.0-nullsafety.0
```

**Review by natebosch — APPROVED** [review]

**Comment by mit-mit** [comment]
@natebosch @jakemac53 what's the story about the needed dependency override?

**Comment by natebosch** [comment]
> what's the story about the needed dependency override?

They will be necessary for packages that are making breaking changes are are also circularly dependencies of anything in `dev_dependencies` like `package:test`. As long as we don't need overrides for stuff in `dependencies` it's still OK to publish, and we then can expand the constraints from things that depend on it and remove the override. It sometimes takes a bit of time, but it doesn't introduce much risk.

**Pull request merged**